### ISSUE TITLE
I can't push to an app that my user has been added to

### ISSUE THREAD

**Issue by kevinob11** [mlebkowski/dokku-acl#25]
## Description of problem
I can't push to an app that I have configured acl for

### How reproducible
Reproducible

### Steps to Reproduce

1. Add user to app acl
2. Add `config:set git-upload-pack git-upload-archive git-receive-pack`
3. Attempt to push to app

#### Actual Results
```
User praece does not have permissions to run git-receive-pack on 'app-test', or 'app-test' does not exist
Access denied
```

#### Expected Results
Push to app as expected

## Environment Information
Dokku 0.17.9

### `dokku report` output
```
-----> uname: Linux serverapp-01 4.4.0-157-generic #185-Ubuntu SMP Tue Jul 23 09:17:01 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
-----> memory:
                     total        used        free      shared  buff/cache   available
       Mem:           7949        1747         584         138        5617        5574
       Swap:           975          23         952
-----> docker version:
       Client: Docker Engine - Community
        Version:           19.03.1
        API version:       1.40
        Go version:        go1.12.5
        Git commit:        74b1e89e8a
        Built:             Thu Jul 25 21:21:35 2019
        OS/Arch:           linux/amd64
        Experimental:      false

       Server: Docker Engine - Community
        Engine:
         Version:          19.03.1
         API version:      1.40 (minimum version 1.12)
         Go version:       go1.12.5
         Git commit:       74b1e89e8a
         Built:            Thu Jul 25 21:20:09 2019
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.2.6
         GitCommit:        894b81a4b802e4eb2a91d1ce216b8817763c29fb
        runc:
         Version:          1.0.0-rc8
         GitCommit:        425e105d5a03fabd737a126ad93d62a9eeede87f
        docker-init:
         Version:          0.18.0
         GitCommit:        fec3683
-----> docker daemon info:
       Client:
        Debug Mode: true

       Server:
        Containers: 78
         Running: 24
         Paused: 0
         Stopped: 54
        Images: 126
        Server Version: 19.03.1
        Storage Driver: overlay2
         Backing Filesystem: extfs
         Supports d_type: true
         Native Overlay Diff: true
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: 894b81a4b802e4eb2a91d1ce216b8817763c29fb
WARNING: No swap limit support
        runc version: 425e105d5a03fabd737a126ad93d62a9eeede87f
        init version: fec3683
        Security Options:
         apparmor
         seccomp
          Profile: default
        Kernel Version: 4.4.0-157-generic
        Operating System: Ubuntu 16.04.6 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 2
        Total Memory: 7.763GiB
        Name: server-01
        ID: SLUU:FSYH:CNSK:IKTF:H7AP:SEG2:MZAB:ECFL:A53N:QKP5:OD6L:MBXJ
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false

-----> sigil version: 0.4.0
-----> herokuish version:
       herokuish: 0.5.0
       buildpacks:
         heroku-buildpack-multi     v1.0.0
         heroku-buildpack-ruby      v200
         heroku-buildpack-nodejs    v142
         heroku-buildpack-clojure   v83
         heroku-buildpack-python    v149
         heroku-buildpack-java      v63
         heroku-buildpack-gradle    v28
         heroku-buildpack-scala     v84
         heroku-buildpack-play      v26
         heroku-buildpack-php       v151
         heroku-buildpack-go        v104
         buildpack-nginx            v10
-----> dokku version: dokku version 0.17.9
-----> dokku plugins:
       plugn: 0.3.2
         00_dokku-standard    0.17.9 enabled    dokku core standard plugin
         20_events            0.17.9 enabled    dokku core events logging plugin
         acl                  1.3.0 enabled    dokku plugin that can be used to restrict push privileges for app to certain users
         app-json             0.17.9 enabled    dokku core app-json plugin
         apps                 0.17.9 enabled    dokku core apps plugin
         build-env            0.17.9 enabled    dokku core build-env plugin
         buildpacks           0.17.9 enabled    dokku core buildpacks plugin
         certs                0.17.9 enabled    dokku core certificate management plugin
         checks               0.17.9 enabled    dokku core checks plugin
         common               0.17.9 enabled    dokku core common plugin
         config               0.17.9 enabled    dokku core config plugin
         docker-options       0.17.9 enabled    dokku core docker-options plugin
         domains              0.17.9 enabled    dokku core domains plugin
         enter                0.17.9 enabled    dokku core enter plugin
         git                  0.17.9 enabled    dokku core git plugin
         letsencrypt          0.8.8 enabled    Automated installation of let's encrypt TLS certificates
         logs                 0.17.9 enabled    dokku core logs plugin
         network              0.17.9 enabled    dokku core network plugin
         nginx-vhosts         0.17.9 enabled    dokku core nginx-vhosts plugin
         plugin               0.17.9 enabled    dokku core plugin plugin
         postgres             1.3.1 enabled    dokku postgres service plugin
         proxy                0.17.9 enabled    dokku core proxy plugin
         ps                   0.17.9 enabled    dokku core ps plugin
         repo                 0.17.9 enabled    dokku core repo plugin
         resource             0.17.9 enabled    dokku core resource plugin
         scheduler-docker-local 0.17.9 enabled    dokku core scheduler-docker-local plugin
         shell                0.17.9 enabled    dokku core shell plugin
         ssh-keys             0.17.9 enabled    dokku core ssh-keys plugin
         storage              0.17.9 enabled    dokku core storage plugin
         tags                 0.17.9 enabled    dokku core tags plugin
         tar                  0.17.9 enabled    dokku core tar plugin
         trace                0.17.9 enabled    dokku core trace plugin
```

### `dokku acl:report` output

```
=====> tools-internal acl information
       Acl allowed users:
       Acl global allow command line: 1
       Acl global super user:         default
       Acl global user commands:      config:get logs help version
       Acl global per app commands:   config:set git-upload-pack git-upload-archive git-receive-pack
amrita
gitlab
praece
=====> app-dev acl information
       Acl allowed users:
       Acl global allow command line: 1
       Acl global super user:         default
       Acl global user commands:      config:get logs help version
       Acl global per app commands:   config:set git-upload-pack git-upload-archive git-receive-pack
amrita
gitlab
praece
=====> app-dev-three acl information
       Acl allowed users:
       Acl global allow command line: 1
       Acl global super user:         default
       Acl global user commands:      config:get logs help version
       Acl global per app commands:   config:set git-upload-pack git-upload-archive git-receive-pack
amrita
gitlab
praece
=====> app-dev-two acl information
       Acl allowed users:
       Acl global allow command line: 1
       Acl global super user:         default
       Acl global user commands:      config:get logs help version
       Acl global per app commands:   config:set git-upload-pack git-upload-archive git-receive-pack
gitlab
praece
=====> app-dic acl information
       Acl allowed users:
       Acl global allow command line: 1
       Acl global super user:         default
       Acl global user commands:      config:get logs help version
       Acl global per app commands:   config:set git-upload-pack git-upload-archive git-receive-pack
gitlab
=====> app-live acl information
       Acl allowed users:
       Acl global allow command line: 1
       Acl global super user:         default
       Acl global user commands:      config:get logs help version
       Acl global per app commands:   config:set git-upload-pack git-upload-archive git-receive-pack
gitlab
praece
=====> app-staging acl information
       Acl allowed users:
       Acl global allow command line: 1
       Acl global super user:         default
       Acl global user commands:      config:get logs help version
       Acl global per app commands:   config:set git-upload-pack git-upload-archive git-receive-pack
gitlab
praece
=====> app-test acl information
       Acl allowed users:
       Acl global allow command line: 1
       Acl global super user:         default
       Acl global user commands:      config:get logs help version
       Acl global per app commands:   config:set git-upload-pack git-upload-archive git-receive-pack
=====> health-check acl information
       Acl allowed users:
       Acl global allow command line: 1
       Acl global super user:         default
       Acl global user commands:      config:get logs help version
       Acl global per app commands:   config:set git-upload-pack git-upload-archive git-receive-pack
```

### `ls -lah ~dokku/.dokkurc/` output

```
total 12K
drwxr-xr-x  2 dokku dokku 4.0K Jul 29 10:14 .
drwxr-xr-x 16 dokku dokku 4.0K Jul 26 22:31 ..
-rw-r--r--  1 dokku dokku  233 Jul 29 10:10 acl
```

### How (deb/make/rpm) and where (AWS, VirtualBox, physical, etc.) was Dokku installed?:
Virtual server on Hyper-v, using deb

**Comment by josegonzalez** [comment]
Can you cat the contents of the `acl` file?

**Comment by kevinob11** [comment]
Sure!

```
export DOKKU_ACL_ALLOW_COMMAND_LINE=1
export DOKKU_SUPER_USER="default"
export DOKKU_ACL_USER_COMMANDS="config:get logs help version"
export DOKKU_ACL_PER_APP_COMMANDS="config:set git-upload-pack git-upload-archive git-receive-pack"
```

Separate topic but I had to set my `DOKKU_SUPER_USER` to `default` in order for my SSH user to be able to be an ACL super user. Is that safe? Is that expected?

**Comment by josegonzalez** [comment]
Can you turn on trace mode and gist the deploy output?

The super user should correspond to whatever user is associated with the SSH key you want to be the super user. Maybe that user is named `default`?

**Comment by kevinob11** [comment]
On the ssh comment, to clarify when I say ssh I'm talking about actually ssh'ing into a bash session on the server, not a key actually registered to dokku. I was expecting that the full ssh session would automatically be a super user.

Here is my trace output
```
+ export DOKKU_HOST_ROOT=/home/dokku
+ DOKKU_HOST_ROOT=/home/dokku
+ export DOKKU_DISTRO
++ . /etc/os-release
++ echo ubuntu
+ DOKKU_DISTRO=ubuntu
+ export DOCKER_BIN=docker
+ DOCKER_BIN=docker
+ export DOKKU_IMAGE=gliderlabs/herokuish:latest
+ DOKKU_IMAGE=gliderlabs/herokuish:latest
+ export DOKKU_LIB_ROOT=/var/lib/dokku
+ DOKKU_LIB_ROOT=/var/lib/dokku
+ export PLUGIN_PATH=/var/lib/dokku/plugins
+ PLUGIN_PATH=/var/lib/dokku/plugins
+ export PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ export PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ export PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ export PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ export PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ export DOKKU_SYSTEM_GROUP=dokku
+ DOKKU_SYSTEM_GROUP=dokku
+ export DOKKU_SYSTEM_USER=dokku
+ DOKKU_SYSTEM_USER=dokku
+ export DOKKU_API_VERSION=1
+ DOKKU_API_VERSION=1
+ export DOKKU_NOT_IMPLEMENTED_EXIT=10
+ DOKKU_NOT_IMPLEMENTED_EXIT=10
+ export DOKKU_VALID_EXIT=0
+ DOKKU_VALID_EXIT=0
+ export DOKKU_LOGS_DIR=/var/log/dokku
+ DOKKU_LOGS_DIR=/var/log/dokku
+ export DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ export DOKKU_CONTAINER_LABEL=dokku
+ DOKKU_CONTAINER_LABEL=dokku
+ export DOKKU_GLOBAL_RUN_ARGS=--label=dokku
+ DOKKU_GLOBAL_RUN_ARGS=--label=dokku
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ parse_args git-receive-pack ''\''app-test'\'''
+ declare 'desc=top-level cli arg parser'
+ local next_index=1
+ local skip=false
+ args=("$@")
+ local args
+ for arg in '"$@"'
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ next_index=2
+ for arg in '"$@"'
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ next_index=3
+ return 0
+ args=("$@")
+ skip_arg=false
+ [[ git-receive-pack =~ ^--.* ]]
+ has_tty
+ declare 'desc=return 0 if we have a tty'
++ /usr/bin/tty
++ true
+ [[ not a tty == \n\o\t\ \a\ \t\t\y ]]
+ return 1
+ DOKKU_QUIET_OUTPUT=1
++ id -un
+ [[ dokku != \d\o\k\k\u ]]
+ [[ git-receive-pack =~ ^plugin:.* ]]
+ [[ git-receive-pack == \s\s\h\-\k\e\y\s\:\a\d\d ]]
+ [[ -n git-receive-pack 'app-test' ]]
+ export -n SSH_ORIGINAL_COMMAND
+ [[ git-receive-pack =~ config-* ]]
+ [[ git-receive-pack =~ docker-options* ]]
+ set -f
+ /usr/bin/dokku git-receive-pack ''\''app-test'\'''
+ export DOKKU_HOST_ROOT=/home/dokku
+ DOKKU_HOST_ROOT=/home/dokku
+ export DOKKU_DISTRO
++ . /etc/os-release
++ echo ubuntu
+ DOKKU_DISTRO=ubuntu
+ export DOCKER_BIN=docker
+ DOCKER_BIN=docker
+ export DOKKU_IMAGE=gliderlabs/herokuish:latest
+ DOKKU_IMAGE=gliderlabs/herokuish:latest
+ export DOKKU_LIB_ROOT=/var/lib/dokku
+ DOKKU_LIB_ROOT=/var/lib/dokku
+ export PLUGIN_PATH=/var/lib/dokku/plugins
+ PLUGIN_PATH=/var/lib/dokku/plugins
+ export PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ export PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ export PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ export PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ export PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ export DOKKU_SYSTEM_GROUP=dokku
+ DOKKU_SYSTEM_GROUP=dokku
+ export DOKKU_SYSTEM_USER=dokku
+ DOKKU_SYSTEM_USER=dokku
+ export DOKKU_API_VERSION=1
+ DOKKU_API_VERSION=1
+ export DOKKU_NOT_IMPLEMENTED_EXIT=10
+ DOKKU_NOT_IMPLEMENTED_EXIT=10
+ export DOKKU_VALID_EXIT=0
+ DOKKU_VALID_EXIT=0
+ export DOKKU_LOGS_DIR=/var/log/dokku
+ DOKKU_LOGS_DIR=/var/log/dokku
+ export DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ export DOKKU_CONTAINER_LABEL=dokku
+ DOKKU_CONTAINER_LABEL=dokku
+ export DOKKU_GLOBAL_RUN_ARGS=--label=dokku
+ DOKKU_GLOBAL_RUN_ARGS=--label=dokku
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ parse_args git-receive-pack ''\''app-test'\'''
+ declare 'desc=top-level cli arg parser'
+ local next_index=1
+ local skip=false
+ args=("$@")
+ local args
+ for arg in '"$@"'
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ next_index=2
+ for arg in '"$@"'
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ next_index=3
+ return 0
+ args=("$@")
+ skip_arg=false
+ [[ git-receive-pack =~ ^--.* ]]
+ has_tty
+ declare 'desc=return 0 if we have a tty'
++ /usr/bin/tty
++ true
+ [[ not a tty == \n\o\t\ \a\ \t\t\y ]]
+ return 1
+ DOKKU_QUIET_OUTPUT=1
++ id -un
+ [[ dokku != \d\o\k\k\u ]]
+ [[ git-receive-pack =~ ^plugin:.* ]]
+ [[ git-receive-pack == \s\s\h\-\k\e\y\s\:\a\d\d ]]
+ [[ -n '' ]]
+ dokku_auth git-receive-pack ''\''app-test'\'''
+ declare 'desc=calls user-auth plugin trigger'
+ export SSH_USER=dokku
+ SSH_USER=dokku
+ export SSH_NAME=praece
+ SSH_NAME=praece
+ plugn trigger user-auth dokku praece git-receive-pack ''\''app-test'\'''
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ [[ ! -n '' ]]
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
++ dirname /var/lib/dokku/plugins/enabled/acl/user-auth
+ source /var/lib/dokku/plugins/enabled/acl/internal-functions
++ source /var/lib/dokku/core-plugins/available/common/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ DOKKU_SUPER_USER=default
+ DOKKU_ACL_USER_COMMANDS='config:get logs help version'
+ DOKKU_ACL_PER_APP_COMMANDS='config:set git-upload-pack git-upload-archive git-receive-pack'
+ DOKKU_ACL_PER_SERVICE_COMMANDS=
+ DOKKU_ACL_LINK_COMMANDS=
+ SSH_USER=dokku
+ SSH_NAME=praece
+ shift 2
+ [[ -z config:get logs help version ]]
+ [[ dokku == \r\o\o\t ]]
+ [[ -n default ]]
+ [[ praece == \d\e\f\a\u\l\t ]]
+ CMD=git-receive-pack
+ for allowed in '$DOKKU_ACL_USER_COMMANDS'
+ [[ git-receive-pack == \c\o\n\f\i\g\:\g\e\t ]]
+ for allowed in '$DOKKU_ACL_USER_COMMANDS'
+ [[ git-receive-pack == \l\o\g\s ]]
+ for allowed in '$DOKKU_ACL_USER_COMMANDS'
+ [[ git-receive-pack == \h\e\l\p ]]
+ for allowed in '$DOKKU_ACL_USER_COMMANDS'
+ [[ git-receive-pack == \v\e\r\s\i\o\n ]]
+ for allowed in '$DOKKU_ACL_PER_APP_COMMANDS'
+ [[ git-receive-pack == \c\o\n\f\i\g\:\s\e\t ]]
+ for allowed in '$DOKKU_ACL_PER_APP_COMMANDS'
+ [[ git-receive-pack == \g\i\t\-\u\p\l\o\a\d\-\p\a\c\k ]]
+ for allowed in '$DOKKU_ACL_PER_APP_COMMANDS'
+ [[ git-receive-pack == \g\i\t\-\u\p\l\o\a\d\-\a\r\c\h\i\v\e ]]
+ for allowed in '$DOKKU_ACL_PER_APP_COMMANDS'
+ [[ git-receive-pack == \g\i\t\-\r\e\c\e\i\v\e\-\p\a\c\k ]]
+ [[ -z 'app-test' ]]
+ fn-check-app-acl ''\''app-test'\''' praece
+ declare 'desc=Checks if the current user has an ACL entry for the app'
+ declare 'APP='\''app-test'\''' SSH_NAME=praece
+ local 'ACL_FILE=/home/dokku/'\''app-test'\''/acl/praece'
+ verify_app_name ''\''app-test'\'''
+ dokku_log_fail 'User praece does not have permissions to run git-receive-pack on '\''app-test'\'', or '\''app-test'\'' does not exist'
+ declare 'desc=log fail formatter'
+ echo 'User praece does not have permissions to run git-receive-pack on '\''app-test'\'', or '\''app-test'\'' does not exist'
User praece does not have permissions to run git-receive-pack on 'app-test', or 'app-test' does not exist
+ exit 1
+ return 1
+ dokku_log_fail 'Access denied'
+ declare 'desc=log fail formatter'
+ echo 'Access denied'
Access denied
+ exit 1
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by josegonzalez** [comment]
Weird, your app is coming in with extra quotes on it, so isn't being detected as an existing application...

Seems like we need to strip single-quotes from the name. This normally happens in the git plugin, but as this gets called before that plugin, we run into the bug. I'll take a look at this soon.

**Comment by kevinob11** [comment]
Thanks, of note if I disable acl command filtering then I can push, so I just changed my acl config to:
```
export DOKKU_ACL_ALLOW_COMMAND_LINE=1
export DOKKU_SUPER_USER="default"
# export DOKKU_ACL_USER_COMMANDS="config:get logs help version"
# export DOKKU_ACL_PER_APP_COMMANDS="config:set git-upload-pack git-upload-archive git-receive-pack"
```
and everything works

**Comment by kevinob11** [comment]
Checking back on this, if you point me to the right place I'm happy to try to find some time to PR.

**Comment by robertorubioguardia** [comment]
Hey guys, I am getting this exact same issue. Anyone around to help?

Even when the acl file is properly set up, a user with access granted cant push. 

See how I added the user to the ACL but the report does not show the user as allowed:

```
root@nodo1:~# dokku acl:add robotloco-web pedro-perez
root@nodo1:~# dokku acl:report robotloco-web
pedro-perez
=====> robotloco-web acl information
       Acl allowed users:                                      
       Acl global allow command line: 1                        
       Acl global super user:         admin1                   
       Acl global user commands:      help version             
       Acl global per app commands:   logs urls ps:rebuild ps:restart ps:stop ps:start git-upload-pack git-upload-archive git-receive-pack letsencrypt domains git-hook
```
If i check in the app's folder I can see the user in the acl folder:

```
root@nodo1:/home/dokku/robotloco-web/acl# ls
pedro-perez
```

**Comment by josegonzalez** [comment]
This is how the git plugin does it: https://github.com/dokku/dokku/blob/master/plugins/git/functions#L206

**Comment by josegonzalez** [comment]
And here is where we want to strip quotes: https://github.com/dokku-community/dokku-acl/blob/c2302e69c391d478fcfaa2d005f72a2853ca0f5f/user-auth#L37

**Comment by josegonzalez** [comment]
The thing we want to strip quotes on is `$2`, which is the app name.

**Comment by robertorubioguardia** [comment]
Thanks for the hint. Just got it fixed.

**Comment by josegonzalez** [comment]
@robertorubioguardia pull requests welcome :)

**Comment by josegonzalez** [comment]
Closed by #28.
